server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/variables/VariableDefinitionUtil.java

```diff
@@ -528,24 +524,51 @@ private List<SyntaxError> processRenamesBoundaries(
 
   private ResultWithErrors<Integer> processRenamesClauseAndGetIndex(
       RenameItemNode variable,
-      VariableNameAndLocality renames,
-      List<VariableNode> nodesForRenaming) {
-    List<SyntaxError> errors = ImmutableList.of();
+      List<VariableNameAndLocality> renames,
+      GroupItemNode group) {
+    List<SyntaxError> errors = new ArrayList<>();
+    List<VariableNode> nodesForRenaming =
+            group.getChildren().stream()
+                    .flatMap(Node::getDepthFirstStream)
+                    .filter(hasType(NodeType.VARIABLE))
+                    .map(VariableNode.class::cast)
+                    .collect(Collectors.toList());
     if (renames == null) return new ResultWithErrors<>(-1, errors);
-    String renamesName = renames.getName();
+    String renamesName = renames.get(0).getName();
     int renamesIndex = Iterables.indexOf(nodesForRenaming, it -> renamesName.equals(it.getName()));
     if (renamesIndex != -1) {
       VariableUsageNode variableUsageNode =
```

Comment:
Let's move this code under the if condition to a separate method f.e. `processRanameClause(variable, renames)`

Reply:
updated